### ISSUE TITLE
Debugging ResourceLeakDetector LEAK

### ISSUE THREAD

**Issue by josephglanville** [ktorio/ktor#1856]
**Ktor Version and Engine Used (client or server and name)**
Ktor version 1.2.6, Netty Engine

Seeing these errors after putting a small service into production with Ktor:
```
2020-05-12 02:17:57.419 [nioEventLoopGroup-3-1] ERROR io.netty.util.ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected. See http://netty.io/wiki/reference-counted-objects.html for more information.
```
After looking at https://github.com/ktorio/ktor/issues/574 and checking for how I use the request body I couldn't find anything obvious.
Most of my handlers are get methods and are of the form:
```kotlin
get("/subjects/{subject}/versions/{version}") {
  val subject = parseSubject(call.parameters["subject"]!!)
  val version = parseVersion(call.parameters["version"]!!)
  storage.get(subject, version)?.let { schema ->
    call.respond(schema.asVersionedResponse(version))
  } ?: call.respond(HttpStatusCode.NotFound)
}
```

I have a single post route which is the only route I interact with the request body:
```
post("/subjects/{subject}") {
  val subject = parseSubject(call.parameters["subject"]!!)
  storage.getAll(subject)?.let { schemas ->
    val body = call.receiveStream().readBytes()
    val requestSchema = mapper.readTree(mapper.readTree(body)["schema"].asText())
    schemas.forEachIndexed { idx, schema ->
      if (requestSchema == schema.schema) {
        call.respond(schema.asVersionedResponse(idx + 1))
        return@post
      }
    }
  }
  call.respond(HttpStatusCode.NotFound)
}
```

Unsure where to look to debug further.

**Comment by e5l** [comment]
Hi @josephglanville, could you reproduce the problem with the ktor `1.3.2`?

**Comment by josephglanville** [comment]
I have just upgraded to `1.3.2` and will have to wait a bit to see if it shows up but already I see much better memory usage and what looked like a small leak seems to be gone. Will update once I can confirm this issue no longer persists.

**Comment by josephglanville** [comment]
This looks to be fixed in `1.3.2` sorry for the noise!
